### PR TITLE
CI: verify emails have been received

### PR DESCRIPTION
**Pull request by webstech**
If configured, an email imap session will verify the sent test emails have been received.  This requires, in addition to existing environment variables, gitgitgadget.CIimapHost to be set.

For GitHub workflows, a new secret GGG_IMAP_HOST must be set.

This feature waits one second after sending the email to check for it.  If this is found to be insufficient (tests failing for not found), retry logic can be added.

This has been tested with the ethereal email service.

For local testing, if the `Error: self signed certificate in certificate chain` message is issued, set environment variable `NODE_TLS_REJECT_UNAUTHORIZED=0`.

### CODE REVIEW

File: tests/ci-helper.test.ts [hunk 1]
```diff
@@ -120,7 +125,7 @@ async function setupRepos(instance: string):
     ]);
 
     const { smtpUser, smtpHost, smtpPass, smtpOpts } =
```
Comment (dscho) [inline tests/ci-helper.test.ts:127]: Doesn't this need to be changed to this?

```js
     const { smtpUser, smtpHost, , smtpPass, smtpOpts } =
```

Comment (dscho) [reply tests/ci-helper.test.ts:127]: Ah, I guess not, because `{}` actually means a map, not a tuple.

File: tests/ci-helper.test.ts [hunk 2]
```diff
@@ -158,6 +163,75 @@ async function setupRepos(instance: string):
...
+
+    return connection;
+}
+
+/**
+ * Terminate the imap mail server connection.
+ *
+ * @param connection
+ */
+async function imapDisConnect(connection: ImapSimple): Promise <void> {
```
Comment (dscho) [inline tests/ci-helper.test.ts:194]:
```suggestion
async function imapDisconnect(connection: ImapSimple): Promise <void> {
```

**Review by dscho — APPROVED** [review]
Looks very good!

I have just that one suggestion: `s/DisConnect/Disconnect/`.

File: tests/ci-helper.test.ts (hunk 2 above)
Comment (webstech) [reply tests/ci-helper.test.ts]: Sorry, not what I would have done from scratch.  Changed.

**Review by dscho — APPROVED** [review]

**Comment by dscho** [comment]
Thank you!!!

**Pull request merged**